backend/session/logind.c

```diff
@@ -241,6 +241,30 @@ static bool take_control(struct logind_session *session) {
 	return ret >= 0;
 }
 
+static bool set_type(struct logind_session *session) {
+	int ret;
+	sd_bus_message *msg = NULL;
+	sd_bus_error error = SD_BUS_ERROR_NULL;
+
+	ret = sd_bus_call_method(session->bus, "org.freedesktop.login1",
+		session->path, "org.freedesktop.login1.Session", "SetType",
+		&error, &msg, "s", "wayland");
+	if (ret < 0) {
+		wlr_log(WLR_ERROR, "Failed to set logind session type for session: %s",
+			error.message);
+	}
+
+	sd_bus_error_free(&error);
+	sd_bus_message_unref(msg);
+
+	ret = setenv("XDG_SESSION_TYPE", "wayland", 1);
```

Comment:
We shouldn't set `XDG_SESSION_TYPE` if `SetType` failed, or the two sources of truth will become out-of-sync.

Reply:
Makes sense definitely for v246, which I guess isn't far away, but on v245 is it worth setting `XDG_SESSION_TYPE` independently? If I'm understanding this right, the only reason `SetType` would realistically fail is if it isn't present on v245, but is it worth setting `XDG_SESSION_TYPE` then? If not then obviously just returning after the `SetType` call is fine.

Reply:
> If not then obviously just returning after the SetType call is fine.

Yup!

Reply:
OK cool.